### ISSUE TITLE
Create basic TSV file of the ontology

### ISSUE THREAD

**Issue by lpalbou** [geneontology/pipeline#161]
We have had a user request to provide a CSV/TSV file with 3 columns:

| GO id | GO label | GO namespace |
| ----- | -------- | --------------- |

Which few additional columns would we want ? 
- [ ] obsolete
- [ ] synonym
- [ ] alt ids
- [ ] definition

It could also be a general export of any ontology in TSV (see https://github.com/ontodev/robot/issues/459)

**Comment by kltm** [comment]
Also see https://github.com/ontodev/robot/issues/459

**Comment by jamesaoverton** [comment]
If you're in a rush, this should be easy enough to do right now with SPARQL and [`robot query`](http://robot.obolibrary.org/query).

**Comment by tamuanand** [comment]
I was curious if this available as a monthly dump on the GO website

**Comment by balhoff** [comment]
I think we'll provide this with the new `robot export` command (mentioned above: https://github.com/ontodev/robot/issues/459). I expect that will be released in a few (or several) weeks, and then we can update our release pipeline.

**Comment by jamesaoverton** [comment]
I've thought that `robot export` was ready for release three times already, and each time somebody found a major bug. It would be great if you can test ROBOT `master` now. I don't plan any more changes to it.

**Comment by tamuanand** [comment]
Hi all,

I was wondering if this is now available

Thanks

**Comment by jamesaoverton** [comment]
I plan to make an official release of ROBOT this week -- apologies for the delay. In the meantime, the development version is available here: https://build.obolibrary.io/job/ontodev/job/robot/job/master/

**Comment by lpalbou** [comment]
@jamesaoverton that would create the TSVs from OBO/OWL ?

Do you have a mockup of that file ?

**Comment by jamesaoverton** [comment]
The ROBOT `export` command is highly customizable. Here’s the documentation: http://robot.obolibrary.org/export

I think that any of the columns mentioned at the top of this issue can be exported in this way.

**Comment by tamuanand** [comment]
@jamesaoverton 
Is there a TLDR section with do this, then do this canned query/command to get the TSV/CSV files being asked

I understand I can look up the github page and the tutorial-command.pdf

In general, having a generic TSV/CSV file (id,name,namespace,def,is_obsolete, synonym, alt_ids) will be always useful

**Comment by tamuanand** [comment]
> I plan to make an official release of ROBOT this week -- apologies for the delay. In the meantime, the development version is available here: https://build.obolibrary.io/job/ontodev/job/robot/job/master/

Hi @jamesaoverton  - I wanted to check if the file is available. Thanks in advance

**Comment by jamesaoverton** [comment]
Yes, I released this two weeks ago, but I forgot to update this issue. I apologize.

https://github.com/ontodev/robot/releases/tag/v1.7.0

**Comment by tamuanand** [comment]
> Yes, I released this two weeks ago, but I forgot to update this issue. I apologize.
> 
> https://github.com/ontodev/robot/releases/tag/v1.7.0

Thanks @jamesaoverton 

How do I use it to get to what I want  (without having to install new tools/software)  - I (and most others on this thread) just need a 3/4 column tab/comma separated file that has something like this

```

id | name | namespace | def
GO:0000019 | regulation of mitotic recombination | BP | Any process that modulates the frequency, rate or extent of DNA recombination during mitosis.
```

Suggestion - It would be better to provide a robot exported dump with every release and make it available via the downloads section on the GO website

**Comment by jamesaoverton** [comment]
@balhoff It looks like this issue is assigned to you. If ROBOT `export` doesn't work as expected, please let me know.

**Comment by tamuanand** [comment]
Also, the instructions here - http://robot.obolibrary.org/ - suggest to use a script from **ontodev** - Is this correct

```
Download the robot.jar file from the latest release.
Save the ROBOT shell script.
OR enter curl https://raw.githubusercontent.com/ontodev/robot/master/bin/robot > robot in the same directory as robot.jar to download it from the terminal.
```

Also, as suggested in earlier message, it would be beneficial if you can have a TLDR (too long didnt read)

As I understand from this - http://robot.obolibrary.org/export, are these steps correct:
```
1. download the robot.jar and the shell script (ontodev OR somewhere else)
2. download go.owl from here - http://purl.obolibrary.org/obo/go.owl  (or will it work with go-basic.obo)
3. use the robot export - http://robot.obolibrary.org/export#columns
4. Not sure how to get the columms I want as laid out in https://github.com/geneontology/pipeline/issues/161#issuecomment-674260573
```

Thanks @jamesaoverton @balhoff

**Comment by balhoff** [comment]
@tamuanand we will be adding a TSV based on `robot export` to the ontology release; however it isn't yet implemented. If you want the definition, the property is `http://purl.obolibrary.org/obo/IAO_0000115`, not `owl:annotatedTarget`.

**Comment by tamuanand** [comment]
> @tamuanand we will be adding a TSV based on `robot export` to the ontology release; however it isn't yet implemented. If you want the definition, the property is `http://purl.obolibrary.org/obo/IAO_0000115`, not `owl:annotatedTarget`.

Hi @balhoff  - could you tell me the code to use to get the 4 things I want starting from the go.owl file -  "id | name | namespace | def" 

Also, the robot script uses a ```ontodev``` version - is that expected?

**Comment by balhoff** [comment]
@tamuanand give this a try: `robot export -i go.owl --header "ID|LABEL|has_obo_namespace|definition" --export table.tsv`

I tried using the identifiers for those two properties, but had some problems (which I will follow up with ROBOT folks). But it works to use the labels for the properties.

> Also, the robot script uses a ontodev version - is that expected?

`ontodev` is the GitHub organization under which ROBOT is developed: https://github.com/ontodev/robot
